### PR TITLE
RFC 0001: Rabbit RFC Process

### PR DESCRIPTION
**Pull request by NateRoiger**
An RFC for RFCs

Signed-off-by: Nate Roiger <nate.roiger@hpe.com>

### CODE REVIEW

File: docs/rfcs/0001/readme.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+==================================
+
+Rabbit software must be designed in close collaboration our end-users. Part of this process involves open discussion in the form of Request For Comment (RFC) documents. The remainder of this document presents the RFC process for Rabbit.
+
+History & Philosophy
+-------------------
+
+NNF RFC documents are modeled after the long history of IETF RFC documents that describe the internet. The philosophy is captured best in [RFC 3](https://www.rfc-editor.org/rfc/rfc3)
+
+> The content of a NWG note may be any thought, suggestion, etc. related to
```
Comment (billhpe) [inline docs/rfcs/0001/readme.md:11]: We should define NWG

**Review by billhpe — APPROVED** [review]

File: docs/rfcs/0001/readme.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+Rabbit Request For Comment Process
+==================================
+
+Rabbit software must be designed in close collaboration our end-users. Part of this process involves open discussion in the form of Request For Comment (RFC) documents. The remainder of this document presents the RFC process for Rabbit.
```
Comment (roehrich-hpe) [inline docs/rfcs/0001/readme.md:4]: our => with our

File: docs/rfcs/0001/readme.md (hunk 1 above)
Comment (roehrich-hpe) [reply docs/rfcs/0001/readme.md:11]: Distracting bits like that should be elided.   "The content of a [...] note may be any ..."

**Review by roehrich-hpe — APPROVED** [review]

File: docs/rfcs/0001/readme.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+inhibition.
+
+
+When to create an RFC
+---------------------
+New features, improvements, and other tasks that need to source feedback from multiple sources are to be written as Request For Comment (RFC) documents.
+
+Metadata
+--------
+At the start of each RFC there must include a short metadata block that contains information useful for filtering and sorting existing documents. This markdown is not visible inside the document.
```
Comment (bdevcich-hpe) [inline docs/rfcs/0001/readme.md:33]:
```suggestion
At the start of each RFC, there must include a short metadata block that contains information useful for filtering and sorting existing documents. This markdown is not visible inside the document.
```

File: docs/rfcs/0001/readme.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+authors: John Doe <john.doe@company.com>, Jane Doe <jane.doe@company.com>
+state: prediscussion|ideation|discussion|published|committed|abandoned
+discussion: (link to PR, if available)
+----
+```
+
+Creation
+---------
+
+A RFC should be created at the next freely available 4-digit index the GitHub RFC folder. Create a folder for your RFC and write your RFC document as `readme.md` using standard Markdown. Include additional documents or images in the folder if needed.
```
Comment (bdevcich-hpe) [inline docs/rfcs/0001/readme.md:46]:
```suggestion
An RFC should be created at the next freely available 4-digit index in the GitHub RFC folder. Create a folder for your RFC and write your RFC document as `readme.md` using standard Markdown. Include additional documents or images in the folder if needed.
```

File: docs/rfcs/0001/readme.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Push
+----
+Push your changes to your RFC branch
+
+```
+git add --all
```
Comment (bdevcich-hpe) [inline docs/rfcs/0001/readme.md:53]:
```suggestion
git add docs/rfcs/<000X>
```

File: docs/rfcs/0001/readme.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+git commit -s -m "[####]: Your Request For Comment Document"
+git push origin ####
+```
+
+Pull-Request
```
Comment (bdevcich-hpe) [inline docs/rfcs/0001/readme.md:58]: Doesn't need a hyphen.

**Review by bdevcich-hpe — APPROVED** [review]
Thanks for writing this up!

**Review by matthew-richerson — APPROVED** [review]

**Pull request merged**